### PR TITLE
DAOS-2730 mgmt: add error codes to drpc responses

### PR DESCRIPTION
**Pull request by tanabarr**
Include DER error codes to dRPC responses in addition
to request status. Update destroy pool resp type.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/mgmt/srv.c [hunk 1]
```diff
@@ -409,26 +427,31 @@ process_createpool_request(Drpc__Call *drpc_req,
...
+		daos_resp->status->state =
+			MGMT__DAOS_REQUEST_STATUS__STATE__ERR_DAOS;
+		daos_resp->status->der = rc;
+	}
 }
 
 static void
-process_destroypool_request(Drpc__Call *drpc_req, Mgmt__DaosResp *daos_resp)
+process_destroypool_request(Drpc__Call *drpc_req,
+			    Mgmt__DestroyPoolResp *daos_resp)
```
Comment (kjacque) [inline src/mgmt/srv.c]: Just a nit, but the name "daos_resp" is a little weird now that the type isn't DaosResp.

**Comment by liw** [comment]
The srv.proto change makes Mgmt__DaosResp.status a pointer. The C code dereferencing it must first allocate the Mgmt__DaosRequestStatus object. This missing part causes segfaults of daos_server. Reverting this PR fixes the issues for me. Since Tom is on vacation and will be in a F2F meeting next week, could we revert this PR, please?

Also, can we think again to see if there's a better way than making "status" a pointer? At least, adding "der" to DaosResp would do? Or, why do we need all dRPC status, DaosResp.status, plus der statuses?

**Comment by mjmac** [comment]
Let's first see if we can salvage this rather than reverting it. The fact that this patch made it through testing without causing any of the segfaults you've seen indicates that we've got a test coverage gap. @liw, would you please create a ticket for this with some information about how to reproduce the crashes so that we can add a test?

**Comment by mjmac** [comment]
After some discussion with @jolivier23, it seems that this didn't trip any test failures because the regular code paths are not yet using non-PMIx mode. Reverting and trying again later seems like the best choice for now.

**Comment by liw** [comment]
Based on the code, I think one may be able to trigger it by invoking the kill rank command of daos_shell.

**Pull request merged**